test/helpers/anvil-fork.ts

```diff
@@ -0,0 +1,48 @@
+import { spawn, ChildProcessWithoutNullStreams } from 'child_process';
+import * as dotenv from 'dotenv';
+
+dotenv.config();
+
+export class AnvilFork {
+  private pathToAnvil: string;
+  private forkUrl: string;
+  private port: string;
+  private blockNumber: number;
+  private process?: ChildProcessWithoutNullStreams;
+
+  constructor(
+    pathToAnvil: string,
+    forkUrl: string,
+    blockNumber: number,
+    port = '8546',
+  ) {
+    this.pathToAnvil = pathToAnvil;
+    this.forkUrl = forkUrl;
+    this.port = port;
+    this.blockNumber = blockNumber;
+  }
+
+  public start() {
+    const forkBlockCommand = `--fork-block-number=${this.blockNumber}`;
+
+    this.process = spawn(`${this.pathToAnvil}/anvil`, [
+      '-f',
+      this.forkUrl,
+      '-p',
+      this.port,
+      forkBlockCommand,
+      '--block-time',
+      '12',
+      '--auto-impersonate',
+    ]);
+
```

Comment:
While developing the library to run `anvil` in Docker (https://github.com/eddort/anvild), I encountered an issue where when `anvil` is started in Docker with fork mode enabled, an error can occur indicating that the process is already running, but the `anvil` node has not yet synchronized. To solve this problem, I used code from another library (https://github.com/wevm/anvil.js).

Below is a solution that helps with this problem:

```ts
export const waitForServerStdout = (stream: NodeJS.ReadWriteStream) => {
  return new Promise((res) => {
    stream.on("data", (data: any) => {
      const output = data.toString();
      if (output.includes("Listening")) {
        res(true);
      }
    });
  });
};
```